server/org.wso2.developerstudio.eclipse.carbonserver44/src/org/wso2/developerstudio/eclipse/carbonserver44/internal/CarbonServerBehavior44.java

```diff
@@ -214,100 +220,91 @@ protected String getAxis2FilePath() {
+					xmlDocument));
+			InputSource catelineSource = new InputSource(new FileInputStream(
+					catelineXmlDocument));
+			DocumentBuilder builder = DocumentBuilderFactory.newInstance()
+					.newDocumentBuilder();
+			DocumentBuilder catelineBuilder = DocumentBuilderFactory
+					.newInstance().newDocumentBuilder();
+			Document document = builder.parse(xmlDocument);
+			Document catelinaDocument = catelineBuilder
+					.parse(catelineXmlDocument);
+			XPath xPath = factory.newXPath();
+			XPath catelineXPath = factory.newXPath();
 			xPath.setNamespaceContext(cntx);
-			
-			int offSet = Integer.parseInt((String)xPath.evaluate("/Server/Ports/Offset", document, XPathConstants.STRING));
-//			XPathExpression  xPathExpression=xPath.compile("/:Server/:Ports/:ServletTransports/:HTTPS");
-//			String evaluate = xPathExpression.evaluate(inputSource);
+
+			int offSet = Integer.parseInt((String) xPath.evaluate(
+					CarbonServerConstants.SERVER_PORTS_OFFSET_XPATH, document,
+					XPathConstants.STRING));
 			String evaluate = (String) catelineXPath
 					.evaluate(
 							CarbonServerConstants.CATALINA_XPATH_EXPRESSION_FOR_SSL_ENABLED_PORT,
 							catelinaDocument, XPathConstants.STRING);
-			
-			if(!evaluate.equals("")){
-				port = Integer.parseInt(evaluate)+offSet;
-			}else{
-				port = getPortfromTransportXML("https");
+
+			if (!evaluate.equals(EMPTY_STRING)) {
+				port = Integer.parseInt(evaluate) + offSet;
+			} else {
+				port = getPortfromTransportXML(CarbonServerConstants.HTTPS_NAME_TAG);
 			}
 			ports.add(port);
-			inputSource =  new InputSource(new FileInputStream(xmlDocument));
-			evaluate = (String)catelineXPath.evaluate("/Server/Service/Connector[1]/@port", catelinaDocument, XPathConstants.STRING);
+			inputSource = new InputSource(new FileInputStream(xmlDocument));
+			evaluate = (String) catelineXPath.evaluate(
+					CarbonServerConstants.SERVER_SERVICE_CONNECTOR_XPATH,
+					catelinaDocument, XPathConstants.STRING);
 
-//			xPathExpression=xPath.compile("/:Server/:Ports/:ServletTransports/:HTTP");
-//			evaluate = xPathExpression.evaluate(inputSource);
-			if(!evaluate.equals("")){
-				port = Integer.parseInt(evaluate)+offSet;
-			}else{
-				port = getPortfromTransportXML("http");
+			if (!evaluate.equals(EMPTY_STRING)) {
+				port = Integer.parseInt(evaluate) + offSet;
+			} else {
+				port = getPortfromTransportXML(CarbonServerConstants.HTTP_NAME_TAG);
 			}
 			ports.add(port);
-			
-		} catch (NumberFormatException e) {
-			e.printStackTrace();
-		} catch (FileNotFoundException e) {
-			e.printStackTrace();
-		} catch (XPathExpressionException e) {
-			e.printStackTrace();
-		} catch (ParserConfigurationException e) {
-			e.printStackTrace();
-		} catch (SAXException e) {
-			e.printStackTrace();
-		} catch (IOException e) {
-			e.printStackTrace();
+
+		} catch (NumberFormatException | XPathExpressionException
+				| ParserConfigurationException | SAXException | IOException e) {
```

Comment:
NumberFormatException is thrown? It is a Runtime Exception. Do we need to handle it?